### PR TITLE
Added reuseable functional test

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR
- Added functional test in re useable workflow 

Boilerplate Changes are in progress https://github.com/flyteorg/boilerplate/pull/52/files
Flytetool Reuseable workflow changes: https://github.com/flyteorg/flytetools/pull/48

Test: https://github.com/flyteorg/flytepropeller/runs/5175177468?check_suite_focus=true
## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: boilerplate/flyte/end2end/end2end.sh [hunk 1]
```diff
@@ -9,11 +9,4 @@ set -e
...
 
-OUT="${DIR}/tmp"
-rm -rf ${OUT}
-git clone https://github.com/flyteorg/flyte.git "${OUT}"
-
-pushd ${OUT}
-
-make end2end_execute
-popd
+python ./boilerplate/flyte/end2end/run-tests.py $VERSION $PRIORITIES
```
Comment (eapolinario) [inline boilerplate/flyte/end2end/end2end.sh:12]: This command should look similar to https://github.com/flyteorg/flytetools/blob/master/flytetester/functional-tests/run.sh#L10. Notice the presence of:
1. the `FLYTE_SDK_USE_STRUCTURED_DATASET=TRUE` prepended to the command.
2. the config file
3. the string "core" to indicate that the script will fail if any of the core tests fail.

File: boilerplate/flyte/end2end/run-tests.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python3
```
Comment (eapolinario) [inline boilerplate/flyte/end2end/run-tests.py:1]: This script will be copied from `flytetools` in the final version, right? I'm not sure how boilerplate works, so I'm asking.

Comment (evalsocket) [reply boilerplate/flyte/end2end/run-tests.py:1]: yes i used the same script

Comment (eapolinario) [reply boilerplate/flyte/end2end/run-tests.py:1]: sorry, I should have been more clear. How are these files under `/boilerplate` updated?

**Review by eapolinario — COMMENTED** [review]
I have just one question, otherwise, it looks good to me.

File: boilerplate/flyte/end2end/run-tests.py (hunk 2 above)
Comment (eapolinario) [reply boilerplate/flyte/end2end/run-tests.py:1]: Followed up offline. We use a workflow to update boilerplate in each repo.

**Review by eapolinario — APPROVED** [review]

**Review by eapolinario — APPROVED** [review]

**Review by pmahindrakar-oss — APPROVED** [review]
